### PR TITLE
Add ability to search blog posts by title, user and content

### PR DESCRIPTION
**Pull request by shavkat-samatov**
https://www.pivotaltracker.com/story/show/135622835

### CODE REVIEW

File: app/models/monologue/post.rb [hunk 1]
```diff
@@ -52,6 +52,36 @@ def self.set_total_pages per_page
...
+    end
+    query
+  end
+
+  def self.search_ids_from_tags(text)
+    Monologue::Tagging.distinct.select(:post_id).joins(:tag).where("monologue_tags.name ILIKE ?", "%#{text}%")
+  end
+
+  def self.search_total_pages
+    search(@text).count
```
Comment (chanko) [inline app/models/monologue/post.rb]: Thoughts on using an attr_accessor so you're not using instance variables in multiple places?

File: app/models/monologue/post.rb [hunk 2]
```diff
@@ -52,6 +52,36 @@ def self.set_total_pages per_page
...
+              .published
+              .where("
+                monologue_posts.title ILIKE :text OR
+                monologue_posts.content ILIKE :text OR
+                users.first_name ILIKE :text OR
+                users.last_name ILIKE :text OR
+                monologue_posts.id IN (#{search_ids_from_tags(text).to_sql})",
+                text: "%#{text}%")
+    if page
+      per_page = 10
```
Comment (chanko) [inline app/models/monologue/post.rb]: Should this be a class constant?

**Comment by shavkat-samatov** [comment]
@chanko thank you for review. I completely agree with your notes. I did what and how I did mostly to match current pagination. I would create a separate ticket/issue to improve/cleanup Monologue::Post class and make those clean and shared.

**Comment by shavkat-samatov** [comment]
@chanko merging as is for now

**Pull request merged**